packages/pigeon/lib/dart_generator.dart

```diff
@@ -248,21 +254,26 @@ void generateDart(DartOptions opt, Root root, StringSink sink) {
             final Field field = klass.fields[index];
             indent.write('..${field.name} = ');
             if (customClassNames.contains(field.dataType)) {
-              indent.add(
-                'pigeonMap[\'${field.name}\'] != null ? ${field.dataType}.decode(pigeonMap[\'${field.name}\']$unwrapOperator) : null',
-              );
+              indent.addln('pigeonMap[\'${field.name}\'] != null');
+              indent.inc(2);
```

Comment:
This code is already covered in the other comment but I don't like naked `indent.inc` or `indent.dec`, that's what `indent.scoped` should be for to make sure each inc is paired with a dec.  Feel free to tweak that function if you need it to do more.